roles/prometheus/templates/prometheus.service.j2

```diff
@@ -54,6 +58,7 @@ RestrictSUIDSGID=true
 
 {% if (ansible_facts.packages.systemd | first).version is version('231', '>=') %}
 ReadWritePaths={{ prometheus_db_dir }}
+WorkingDirectory={{ prometheus_db_dir }}
```

Comment:
when `--storage.tsdb.path` is removed, prometheus is trying to write some tsdb related file to `/` aka root of filesystem, and since by default it's running from `prometheus` use we're getting an error:
```Aug 17 20:26:44 druid-master2-int prometheus[50516]: ts=2023-08-17T20:26:44.773Z caller=main.go:1155 level=error err="opening storage failed: lock DB directory: mkdir /data-agent: read-only file system"```
So defining the `WorkingDirectory` fixes this issue

Reply:
You need to set `--storage.agent.path`.

Reply:
thanks, replace with `--storage.agent.path={{ prometheus_db_dir }}`